src/pin-field/PinField.ts

```diff
@@ -142,6 +157,22 @@ export class PinField extends OmniFormElement {
         this.value = input?.value;
     }
 
+    // When a value is pasted in the input.
+    _onPaste(e: ClipboardEvent) {
+        const input = this._inputElement as HTMLInputElement;
+        const clipboardData = e.clipboardData;
+        let pastedData = clipboardData?.getData('Text');
+
+        if (pastedData && new RegExp('^[0-9]+$').test(pastedData) === false) {
```

Comment:
Wont this only run for non numbers?

Reply:
The function will be removed as the attributeChangedCallback function solves paste scenarios